client/src/components/ContentNavigator/ContentDataProvider.ts

```diff
@@ -244,12 +245,18 @@ class ContentDataProvider
     item: ContentItem,
     name: string,
   ): Promise<Uri | undefined> {
-    if (!(await closeFileIfOpen(item))) {
+    const closing = closeFileIfOpen(item);
+    if (!(await closing)) {
```

Comment:
```suggestion
    const closing = await closeFileIfOpen(item);
    if (!closing) {
```

Reply:
Well I intended to get the variable before promise resolve, so that I can tell if the file was open at line 255.

Reply:
I don't think I understand

Reply:
If I wrote `const closing = await closeFileIfOpen(item);`, then if  `closing === true`, there're 2 possibilities.
One is that the file was not open, the `closeFileIfOpen` function directly returned true, in this case, it should not re-open the file.
The other is, the file was open and the user confirmed save or not save and closed the file, the function returned a promise and resolved to true, in this case, it should re-open the file.
I can't distinguish the 2 cases if I only get awaited true.